server/block/hopper.go

```diff
@@ -128,6 +128,17 @@ func (h Hopper) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.
 	return placed(ctx)
 }
 
+// RedstoneUpdate ...
+func (h Hopper) RedstoneUpdate(pos cube.Pos, w *world.World) {
+	powered := receivedRedstonePower(pos, w)
+	if powered == h.Powered {
+		return
+	}
+
+	h.Powered = powered
```

Comment:
If !powered you should reset the cooldowns as well as short circuiting the Tick() method like mentioned on discord